internal/mode/static/nginx/config/upstreams.go

```diff
@@ -58,14 +68,16 @@ func createUpstream(up dataplane.Upstream) http.Upstream {
 	}
 
 	return http.Upstream{
-		Name:    up.Name,
-		Servers: upstreamServers,
+		Name:     up.Name,
+		ZoneSize: zoneSize,
+		Servers:  upstreamServers,
 	}
 }
 
 func createInvalidBackendRefUpstream() http.Upstream {
 	return http.Upstream{
-		Name: invalidBackendRef,
+		Name:     invalidBackendRef,
+		ZoneSize: ossZoneSize,
```

Comment:
Are we only ever going to have one server here? Can this be smaller?

Reply:
> Are we only ever going to have one server here? Can this be smaller?

yep

Reply:
How small you think we should go? 8 or 16k?

Reply:
>  based on our current math, we could get away with 4k, but I'd rather not cut it that close.

8 seems reasonable then

Reply:
Well, looks like 32k is the smallest nginx will allow.